app/services/analytics_events.rb

```diff
@@ -1332,6 +1332,62 @@ def openid_connect_token(client_id:, user_id:, **extra)
     )
   end
 
+  # Tracks if otp phone validation failed
+  # @param [String] error
+  # @param [String] context
+  # @param [String] country
+  def otp_phone_validation_failed(error:, context:, country:, **extra)
+    track_event(
+      'Twilio Phone Validation Failed',
+      error: error,
+      context: context,
+      country: country,
+      **extra,
+    )
+  end
+
+  # User has been marked as authenticated
+  # @param [String] authentication_type
+  def user_marked_authed(authentication_type:, **extra)
+    track_event(
+      'User marked authenticated',
+      authentication_type: authentication_type,
+      **extra,
+    )
+  end
+
+  # User registration has been hadnded off to agency page
+  # @param [Boolean] ial2
+  # @param [Integer] ialmax
+  # @param [String] service_provider_name
+  # @param [String] page_occurence
+  # @param [String] needs_completion_screen_reason
+  # @param [Array] sp_request_requested_attributes
+  # @param [Array] sp_session_requested_attributes
+  def user_registration_agency_handoff_page_visit(
+      ial2:,
+      ialmax:,
+      service_provider_name:,
+      page_occurence:,
+      needs_completion_screen_reason:,
+      sp_request_requested_attributes:,
+      sp_session_requested_attributes:,
```

Comment:
are all of these params present always? are any of them optional? (ex should any of them have a default of `nil`?)

Reply:
Spot-checking a few callsites and the `analyics_attributes('')` method to see if there's any branching

Reply:
Updated, please check.